syft/pkg/cataloger/java/archive_parser.go

```diff
@@ -408,36 +415,45 @@ func newPackageFromMavenData(pomProperties pkg.PomProperties, pomProject *pkg.Po
 }
 
 func packageIdentitiesMatch(p pkg.Package, parentPkg *pkg.Package) bool {
-	// the name/version pair matches...
-	if uniquePkgKey(&p) == uniquePkgKey(parentPkg) {
-		return true
-	}
+	childMetadata, childOk := p.Metadata.(pkg.JavaMetadata)
```

Comment:
nit: maybe remove `child` from this and `childGroupID`, since we're talking about _this_ package, not a child package.

Reply:
resolved with [0bf6be2](https://github.com/anchore/syft/pull/2033/commits/0bf6be24b5604ef21647616245978ded44ff847f)

good change!